pkg/source/adapter/adapter.go

```diff
@@ -183,6 +183,9 @@ func (a *Adapter) processEntry(ctx context.Context, conn redis.Conn, streamName
 	reply, err := conn.Do("XREADGROUP", "GROUP", groupName, consumerName, "COUNT", count, "BLOCK", blockms, "STREAMS", streamName, xreadID)
 	if err != nil {
 		a.logger.Error("Cannot read from stream", zap.Error(err))
+		if strings.Contains(strings.ToLower(err.Error()), "use of closed network connection") { // Redis has probably been shut down
+			panic("Redis has been shut down")
+		}
```

Comment:
Is there a more graceful way to handle this error? From the original issue, it seems like the proposed solution is:
>  Eventing-redis should reopen a socket when the error is "use of closed network connection".

I imagine that with this panicking, it will eventually be restarted and that would reopen the socket. But, I feel like we could handle this more gracefully. WDYT?